extensions/iceberg/src/main/java/io/deephaven/iceberg/util/SortOrderProvider.java

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+/**
+ * A specification for providing {@link org.apache.iceberg.SortOrder} while writing to an iceberg table.
+ */
+public interface SortOrderProvider {
+
+    // Static factory methods for creating SortOrderProvider instances
+
+    /**
+     * Do not sort the data while writing new data to the iceberg table.
+     */
+    static SortOrderProvider disableSorting() {
+        return SortOrderProviderInternal.DisableSorting.getInstance();
+    }
+
+    /**
+     * Use the default {@link org.apache.iceberg.Table#sortOrder()} of the table while writing new data.
+     */
+    static SortOrderProvider useTableDefault() {
+        return SortOrderProviderInternal.TableDefaultSortOrderProvider.getInstance();
+    }
+
+    /**
+     * Use the sort order with the given ID to sort new data while writing to the iceberg table.
+     */
+    static SortOrderProvider fromSortId(final int id) {
+        return new SortOrderProviderInternal.IdSortOrderProvider(id);
+    }
+}
```

Comment:
Another option might be "I want to these specific fields, even though it may not have a valid sortorder id (we won't actually write it down)"

```java
static SortOrderProvider fromSortOrder(SortOrder order) { ... }
```

You could also couple this with allowing the caller to be explicit what they actually want to write down as the sort order id.

```java
public interface SortOrderProvider {

  SortOrderProvider as(final int sortOrderId);
}
```

This would allow you to do something like:

```java
SortOrderProvider.fromSortOrder(FOO_BAR_BAZ).as(FOO_BAR)
```

and we would need to add `org.apache.iceberg.SortOrder#satisfies` checks.

This seems like a nice generalization that could be useful?